src/plugins/sync-plugin.js

```diff
@@ -178,6 +180,39 @@ export const getRelativeSelection = (pmbinding, state) => ({
   head: absolutePositionToRelativePosition(state.selection.head, pmbinding.type, pmbinding.mapping)
 })
 
+const getElementFromTextNode = node => {
+  if (dom.checkNodeType(node, dom.ELEMENT_NODE)) {
+    return node
+  }
+  return node.parentElement
+}
+
+const isDomSelectionInView = () => {
+  const selection = window.getSelection()
```

Comment:
This should be `view._root.getSelection()` so it works properly in web components (they have a different document root).

Reply:
That is a good one. Didn't think about the web components. Moved this function also to the ProsemirrorBinding class.